tasks/borg-client.yml

```diff
@@ -51,24 +51,36 @@
     user: "{{ ansible_user_id }}"
     key: "{{ sshkey.stdout }}"
     key_options: 'command="cd {{ item.pool }}/{{ inventory_hostname }};/usr/local/bin/borg1 serve {% if borgbackup_appendonly %}--append-only {% endif %} --restrict-to-path {{ item.pool }}/{{ inventory_hostname }}",no-port-forwarding,no-X11-forwarding,no-pty,no-agent-forwarding,no-user-rc'
-    path: "/tmp/rsync.net-{{ item.fqdn }}-authkeys"
+    path: "/tmp/authkeys-{{ item.type }}-{{ item.fqdn }}-authkeys"
     manage_dir: no
   delegate_to: localhost
   become: no
   when: item.type == 'rsync.net'
   with_items: "{{ borgbackup_servers }}"
   register: authkeys
 
-- name: client | upload local authorized_keys to rsync.net
-  raw: scp /tmp/rsync.net-{{ item.fqdn }}-authkeys {{ item.user }}@{{ item.fqdn }}:.ssh/authorized_keys
+- name: client | modify local hetzner authorized_keys
+  authorized_key:
+    user: "{{ ansible_user_id }}"
+    key: "{{ sshkey.stdout }}"
```

Comment:
I think you forgot the key options? This means any client will have full access to your entire account.